src/shared/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,81 @@
+// this rule gives trouble on forward refed components
+/* eslint-disable react/display-name */
+import React from 'react'
+import Icon from '../Icon'
+import { Checkmark, Container, InnerContainer, Input } from './Checkbox.styles'
+
+type HTMLCheckboxProps = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+export interface CheckboxProps extends Omit<HTMLCheckboxProps, 'value' | 'onChange' | 'checked' | 'multiple' | 'ref'> {
+  value: boolean
+  disabled?: boolean
+  indeterminate?: boolean
+  error?: boolean
+  className?: string
+  onChange?: (value: boolean) => void
+  onFocus?: (event: React.FocusEvent<HTMLDivElement>) => void
+  onBlur?: (event: React.FocusEvent<HTMLDivElement>) => void
+}
+
+export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  ({ value, indeterminate, onChange, disabled = false, onFocus, onBlur, error = false, className, ...props }, ref) => {
+    const isIndeterminate = indeterminate || false
+    const isSelected = !!value
+    const [isHovered, setIsHovered] = React.useState<boolean>(false)
+    const [isFocused, setIsFocused] = React.useState<boolean>(false)
+
+    return (
+      <Container
+        selected={value}
+        disabled={disabled}
+        isFocused={isFocused}
+        isHovered={isHovered}
+        error={error}
+        onMouseEnter={() => {
```

Comment:
Not super sure, but are these onMouseEnter and onMouseLeave required? Could we just use CSS here?

Reply:
But actually, it's cleaner this way coz otherwise in the styles we would have to extract the Container Hover/Focused css into a separate function to be reused

Reply:
Still think it's much better to remove them. I managed to change it by myself and I think it works well. 
```js
export const Container = styled.div<CheckboxStateProps>`
  position: relative;
  width: max-content;
  padding: ${sizes(2)};
  margin: -${sizes(2)};
  cursor: pointer;
  border-radius: 100%;
  color: ${colors.gray[300]};
  transition: background ${transitions.timings.loading} ${transitions.easing};
  ${({ isFocused, selected, disabled }) =>
    isFocused &&
    `background: ${colors.gray[800]};
    ${selected && `background: ${colors.blue[900]};`}
    ${disabled && `background: transparent;`}`}

  ${({ disabled }) =>
    !disabled &&
    css`
      :hover {
        background: ${colors.gray[800]};
      }
    `};
`
```

Reply:
Still think it's much better to remove `onMouseEnter` and `onMouseLeave`. 
```js